models/buffer.go

```diff
@@ -32,19 +44,31 @@ type Buffer struct {
 }
 
 // NewBuffer returns a new empty Buffer with the given capacity.
-func NewBuffer(name string, alias string, capacity int) *Buffer {
+func NewBuffer(name string, alias string, capacity int, strategy string, path string) (Buffer, error) {
+	bm := NewBufferMetrics(name, alias, capacity)
+
+	switch strategy {
+	case "", "memory":
+		return NewMemoryBuffer(capacity, bm)
+	case "disk":
+		return NewDiskBuffer(name, path, bm)
+	case "overflow":
+		// todo implementme
+		// todo log currently unimplemented
+		return NewMemoryBuffer(capacity, bm)
+	}
+
+	// todo log invalid buffer strategy configuration provided, falling back to memory
+	return NewMemoryBuffer(capacity, bm)
```

Comment:
No. Please fail here with
```suggestion
	return nil, fmt.Errorf("invalid buffer strategy %q", strategy)
```
Don't try to be clever as this leads to unexpected behavior in a very critical area!